### PR TITLE
UAR-915-BO-OLE-GOV-principal-address-display

### PR DESCRIPTION
**Pull request by olaych**
…nnot be displayed'

### JIRA link
https://companieshouse.atlassian.net/browse/UAR-915

### Change description
Principal address value shown as "Cannot be displayed"

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: views/includes/check-your-answers/beneficial-owner-gov.html [hunk 1]
```diff
@@ -3,8 +3,12 @@
 {% set bogStartDate = dateMacros.formatDate(bog.start_date["day"], bog.start_date["month"], bog.start_date["year"]) %}
 
 {% set bogFormattedPrincipalAddressHtml %}
-  {% set address = bog.principal_address %}
-  {% include "includes/display_address.html" %}
+  {% if (not bog.principal_address.length > 0  and pageParams.noChangeFlag) %}
```
Comment (AndreKramer7654) [inline views/includes/check-your-answers/beneficial-owner-gov.html:6]: why length > 0? bog.principal_address is an object.

File: views/includes/check-your-answers/beneficial-owner-other.html [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {% import "includes/date-macros.html" as dateMacros %}
 
 {% set bocStartDate = dateMacros.formatDate(boc.start_date["day"], boc.start_date["month"], boc.start_date["year"]) %}
-
 {% set bocFormattedPrincipalAddressHtml %}
-  {% if not boc.principal_address and pageParams.noChangeFlag %}
+  {% if (not boc.principal_address.length > 0  and pageParams.noChangeFlag) %}
```
Comment (AndreKramer7654) [inline views/includes/check-your-answers/beneficial-owner-other.html:5]: as above

File: views/includes/check-your-answers/beneficial-owner-gov.html (hunk 1 above)
Comment (AndreKramer7654) [reply views/includes/check-your-answers/beneficial-owner-gov.html:6]: {%if not bog.principle_address and pageParams.noChangeFlag %}

Comment (olaych) [reply views/includes/check-your-answers/beneficial-owner-gov.html]: @AndreKramer7654 this was the syntax that existed for the bo other below before that did not work. 

Yes, it is an object meant to user var|length > 0 

https://github.com/mozilla/nunjucks/releases/tag/v2.5.0

File: views/includes/check-your-answers/beneficial-owner-other.html (hunk 2 above)
Comment (olaych) [reply views/includes/check-your-answers/beneficial-owner-other.html]: As above.

File: views/includes/check-your-answers/beneficial-owner-gov.html [hunk 3]
```diff
@@ -3,8 +3,12 @@
 {% set bogStartDate = dateMacros.formatDate(bog.start_date["day"], bog.start_date["month"], bog.start_date["year"]) %}
 
 {% set bogFormattedPrincipalAddressHtml %}
-  {% set address = bog.principal_address %}
-  {% include "includes/display_address.html" %}
+  {% if (not bog.principal_address|length > 0  and pageParams.noChangeFlag) %}
```
Comment (bmcgreevyCH) [inline views/includes/check-your-answers/beneficial-owner-gov.html:6]: These probably read better the other way around, at first glance it looks like NOT applies to both variables... but it still works so I won'y delay things :-)

File: views/includes/check-your-answers/beneficial-owner-other.html [hunk 4]
```diff
@@ -1,9 +1,8 @@
 {% import "includes/date-macros.html" as dateMacros %}
 
 {% set bocStartDate = dateMacros.formatDate(boc.start_date["day"], boc.start_date["month"], boc.start_date["year"]) %}
-
 {% set bocFormattedPrincipalAddressHtml %}
-  {% if not boc.principal_address and pageParams.noChangeFlag %}
+  {% if (not boc.principal_address|length > 0  and pageParams.noChangeFlag) %}
```
Comment (bmcgreevyCH) [inline views/includes/check-your-answers/beneficial-owner-other.html:5]: Same as above

**Review by bmcgreevyCH — APPROVED** [review]

File: views/includes/check-your-answers/beneficial-owner-gov.html (hunk 3 above)
Comment (olaych) [reply views/includes/check-your-answers/beneficial-owner-gov.html]: Hahahaha okay, took out the not and compared the length to 0.

**Pull request merged**